### ISSUE TITLE
There is no longer a good way to override/extend surveyor helper methods.

### ISSUE THREAD

**Issue by mgurley** [NUARIG/surveyor#213]
Using surveyor (0.20.0):  There is no longer a good way to override/extend surveyor helper methods.  Surveyor used to allow you to create helpers/surveyor_helper_extensions.rb  to override/extend methods in the SurveyorHelper module like so:

```
module SurveyorHelperExtensions
  def self.included(base) # :nodoc:
    base.send(:include, InstanceMethods)
    base.class_eval do
      # Same as typing in the module
      # alias_method_chain :question_number_helper, :sauce
      def next_section()
        if (@sections.last == @section)
          concat(submit_tag("Confirm", :name => "finish", :id => 'finish_button', :class => 'button positive'))
        else
          concat(submit_tag("Next&raquo;", :name => "section[#{@sections[@sections.index(@section)+1].id}]", :id => 'next_section_button'))
        end
      end

      def previous_section
        submit_tag("&laquo; Previous", :name => "section[#{@sections[@sections.index(@section)-1].id}]", :id => 'previous_section_button') unless @sections.first == @section
      end

      def menu_button_first
        link_to "Patient", request.env['HTTP_REFERER']
      end
    end
  end

  module InstanceMethods
    # def question_number_helper_with_sauce(number)
    #   question_number_helper_without_sauce(number) + "Extra sauce"
    # end
  end
end
```

This code was switched on by setting the following in config/initializers/surveyor.rb like so:

```
Surveyor::Config.run do |config|
  config['extend'] = %w(surveyor_helper)
end
```

And the actual surveyor_helper.rb loaded it based on the configuration like so:

```
module SurveyorHelper
  #...Removed for brevity
  # Extending surveyor
  include SurveyorHelperExtensions if Surveyor::Config['extend'].include?("surveyor_helper")
end
```

Now there is now way to extend a method in the SurveyorHelper and the only way to override one that I have found is to do the following:
- Extend the SuveyorController in the normal recommended fashion.
- Add a method to the SurveyorController with the same name as the helper method you want to override.
- Expose the method as a helper method.

For example:

```
module SurveyorControllerCustomMethods
  def self.included(base)
  end

  def q_text(obj)
    return image_tag(obj.text) if obj.is_a?(Question) and obj.display_type == "image"
    return obj.text if obj.is_a?(Question) and (obj.dependent? or obj.display_type == "label" or obj.part_of_group?)
    "#{obj.text}"
  end
end

class SurveyorController < ApplicationController
  include Surveyor::SurveyorControllerMethods
  include SurveyorControllerCustomMethods

  helper_method :q_text
end
```

I have often needed to override/extend SurveyorHelper methods.  It would be nice to have a more elegant solution.

**Comment by mgurley** [comment]
Actually my workaround does not work because a controller methods does not have access to all the usual helper methods: tag and form helpers...So my example would not work for a question with a display_type == 'image'.  It would seem that the only workaround is to copy the SurveyorHelper module locally and manage upstream changes as they occur.

**Comment by sarahmei** [comment]
Do you guys have an ETA for this? Currently we're just copying the entire survey helper into our app and changing the one line we need to be different, which is suboptimal for lots of reasons.

Thanks!

**Comment by mgurley** [comment]
Sarah,

I will be looking at resolving this issue sometime within this month.

Mike

**Comment by sarahmei** [comment]
Thanks Mike, I <3 this gem and really appreciate your work on it.

**Comment by yoon** [comment]
Closed by #279.
